lightning-invoice/src/payment.rs

```diff
@@ -256,6 +272,11 @@ pub trait Router<S: Score> {
 		&self, payer: &PublicKey, route_params: &RouteParameters, payment_hash: &PaymentHash,
 		first_hops: Option<&[&ChannelDetails]>, scorer: &S
 	) -> Result<Route, LightningError>;
+
+	/// Builds a [`Route`] from `payer` along the given path. 
+	fn build_route_from_hops(
```

Comment:
Hm, but isn't the interface in `InvoicePayer` supposed to wrap more functionality in a single method call, such as `pay_invoice`, `pay_pubkey`? If `send_route_along_path` should take a `Route` I honestly don't see it's utility over simply using `send_probing_payment` in `ChannelManager` directly, i.e., I'd just leave it out of `InvoicePayer` entirely?

Reply:
Alright. I agree that eventually we'd want to give a range of probing tools to the user, the main one probably being a method probing a particular *channel* rather than a full path. I'd however still argue probing a path should be part of that toolbox. That said I'm fine with working out the details of the API in a follow-up and here only provide `ChannelManager`'s `send_probe`.